torchtune/training/quantization.py

```diff
@@ -29,6 +32,27 @@
     Int8DynActInt4WeightQATQuantizerModuleSwap,
 )
 
+from torchtune.modules import TransformerDecoder
+from torchtune.modules.low_precision._utils import _get_torchao_version
+from torchtune.modules.peft import DoRALinear, LoRALinear
+from torchtune.utils._version import torch_version_ge
+
+
+_TORCHAO_VERSION, _ = _get_torchao_version()
+
+_SUPPORTS_INT8_MIXED_PRECISION_TRAINING = (
```

Comment:
how bad is the performance if flex attention if off? I am thinking if this should be part of the requirement. If so, then we can just include this import guard in the check: https://github.com/pytorch/torchtune/blob/0a3762d058fd9d860606a3a6bbebf71ac593ab5c/torchtune/utils/_import_guard.py#L11

Maybe thats a bad idea, we can raise a warning instead: Use compile and dataset.packed for best results.